### ISSUE TITLE
Picking up learning rate where left off in resume from and auto resume

### ISSUE THREAD

**Issue by hssandriss** [open-mmlab/mmsegmentation#1800]
Hello, 

Thanks for the great work on the library. 
I am trying to resume model training but somehow the learning rate scheduler (polynomial) doesn't continue where it the checkoint is saved. Please have a look at the figure. 

I am using epoch based runner for 120 epochs with config as follows:
```
log_config = dict(
    interval=50,
    by_epoch=True,
    hooks=[
        dict(type='TextLoggerHook', by_epoch=True),
        dict(type='TensorboardLoggerHook')
    ])
log_level = 'INFO'
load_from = None
resume_from = 'work_dirs/deeplabv3_r50-d8_720x720_70e_cityscapes_20220719151219_TEST/epoch_70.pth'
workflow = [('train', 1)]
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0005)
optimizer_config = dict()
lr_config = dict(policy='poly', power=0.9, min_lr=0.0001, by_epoch=False)
runner = dict(type='EpochBasedRunner', max_epochs=120)
checkpoint_config = dict(by_epoch=True, interval=5)
evaluation = dict(interval=10, metric='mIoU', pre_eval=True)
```

![image](https://user-images.githubusercontent.com/51092741/179955133-0acbe667-c933-42ea-9cd7-9195e9e115ab.png)

 Last learning rate is 1.126e-04 and first learning rate after resume 4.598e-03
I checked in the checkpoint and found that the "correct" next learning rate is there (1.004e-4) as in the screenschot below.
![image](https://user-images.githubusercontent.com/51092741/179959152-f9c4f341-d610-4cc8-8004-9b6e69a48f0e.png)
